doc/development/source/tutorial-utilities.rst

```diff
@@ -0,0 +1,146 @@
+Progress Bar
+************
+
+Operations that take more than a split second, indicate their
+progress with a progress bar.
+
+.. image:: images/learningcurve.png
+
+or in the title bar of the widget's window. There, Orange also
+reports on the estimated time of completion of the task:
+
+.. image:: images/learningcurve-output.png
+
+There are three mechanisms for reporting on progress.
+
+Context handlers
+................
+
+The recommended and the simplest approach is to wrap the code that
+uses a progress bar into a context handler
+:obj:`~Orange.widgets.widgetOWWidget.progressBar`. If the operation
+consists of known number of steps that take similar time, the context
+handler can be used as follows::
+
+    with self.progressBar(n) as progress:
+        for k in range(n):
+            self.do_some_processing()
+            progress.advance()
+
+The `progress.advance()` method will advance the progress bar by `1/n`.
+
+In general, the argument for `self.progressBar` represents the total
+towards which the progress bar advances. Method `progress.advance()` accepts
+an argument representing the advance. ::
+
+    with self.progressBar(n) as progress:
+        for k in s:
+            self.do_some_processing()
+            progress.advance(i)
+
+In this case, the values of `i` should sum up to `n`.
+
+If the argument to `self.progressBar`, `n` is omitted, it defaults to 1, so
```

Comment:
Should be:

> If the argument, `n`, is omitted

cf. [Rule 5](http://www.grammarbook.com/punctuation/commas.asp).